### PR TITLE
Interpret a blob of memory as a rar file for fuzzing.

### PR DESCRIPTION
**Pull request by aawc**
This is experimental but is expected to increase fuzzing speed.

I am going to run with this configuration for 1 week and then compare the performance from previous week to see if it is worth doing this.

For more discussion, see google#951

### CODE REVIEW

File: projects/unrar/unrar_fuzzer.cc [hunk 1]
```diff
@@ -9,19 +9,13 @@ extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
...
-  file.write(reinterpret_cast<const char *>(data), size);
-  file.close();
 
   std::unique_ptr<CommandData> cmd_data(new CommandData);
   cmd_data->ParseArg(const_cast<wchar_t *>(L"-p"));
   cmd_data->ParseArg(const_cast<wchar_t *>(L"x"));
   cmd_data->ParseDone();
   std::wstring wide_filename(filename.begin(), filename.end());
+  cmd_data->SetArcInMem(const_cast<unsigned char *>(data), size);
   cmd_data->AddArcName(wide_filename.c_str());
```
Comment (alex) [inline projects/unrar/unrar_fuzzer.cc]: Can the filename be entirely removed now, or is it still required?

Comment (aawc) [reply projects/unrar/unrar_fuzzer.cc]: Unfortunately, it is still required. Once we decide that we are definitely going down this path (in-memory), I can convince the maintainer to support it better i.e. not have to set filename at all,

Comment (inferno-chromium) [reply projects/unrar/unrar_fuzzer.cc]: Does filename have to be unique, otherwise, we can just set it to a default like temp.rar ?

Comment (aawc) [reply projects/unrar/unrar_fuzzer.cc]: It doesn't have to be but I agree that having a fixed name now is possible and faster so let me change that.

Comment (aawc) [reply projects/unrar/unrar_fuzzer.cc]: Done.

**Pull request merged**